.github/workflows/flatpak.yml

```diff
@@ -19,35 +24,91 @@ env:
+        with:
+          bundle: obs-studio-${{ github.sha }}.flatpak
+          manifest-path: CI/flatpak/com.obsproject.Studio.json
+          cache-key: flatpak-builder-${{ github.sha }}
+
+  publish:
+    name: Publish to Flathub
     runs-on: [ubuntu-latest]
+    if: github.event_name == 'release'
+    env:
+      FLATPAK_BUILD_FOLDER: flatpak_app/files/share/
     container:
       image: bilelmoussaoui/flatpak-github-actions:kde-5.15-21.08
       options: --privileged
     steps:
-    - name: 'Check for Github Labels'
-      if: github.event_name == 'pull_request'
-      shell: bash
-      run: |
-        LABELS_URL="$(echo ${{ github.event.pull_request.url }} | sed s'/pulls/issues/')"
-        LABEL_FOUND="$(curl -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" "${LABELS_URL}/labels" | sed -n 's/.*"name": "\(.*\)",/\1/p' | grep 'Seeking Testers' || true)"
-        if [ "${LABEL_FOUND}" = "Seeking Testers" ]; then
-          echo "SEEKING_TESTERS=1" >> $GITHUB_ENV
-        else
-          echo "SEEKING_TESTERS=0" >> $GITHUB_ENV
-        fi
-
-    - name: Checkout
-      uses: actions/checkout@v2.3.3
-      if: success() && (github.event_name != 'pull_request' || env.SEEKING_TESTERS == '1')
-      with:
-        submodules: 'recursive'
-
-    - name: Build Flatpak Manifest
-      uses: bilelmoussaoui/flatpak-github-actions/flatpak-builder@v3
-      if: success() && (github.event_name != 'pull_request' || env.SEEKING_TESTERS == '1')
-      with:
-        bundle: obs-studio-${{ github.sha }}.flatpak
-        manifest-path: CI/flatpak/com.obsproject.Studio.json
-        cache-key: flatpak-builder-${{ github.sha }}
+      - name: Checkout
+        uses: actions/checkout@v2.3.3
+        with:
+          submodules: 'recursive'
+
+      - name: Build Flatpak Manifest
+        uses: bilelmoussaoui/flatpak-github-actions/flatpak-builder@master
+        with:
+          bundle: obs-studio-${{ github.sha }}.flatpak
+          manifest-path: CI/flatpak/com.obsproject.Studio.json
+          cache-key: flatpak-builder-${{ github.sha }}
+          mirror-screenshots-url: https://dl.flathub.org/repo/screenshots
+
+      - name: Validate AppStream
+        shell: bash
+        run: |
+          cd ${{ env.FLATPAK_BUILD_FOLDER }}
+          appstream-util validate appdata/com.obsproject.Studio.appdata.xml
+
+      - name: Verify icon and metadata in app-info
+        shell: bash
+        run: |
+          cd ${{ env.FLATPAK_BUILD_FOLDER }}
+          test -f app-info/icons/flatpak/128x128/com.obsproject.Studio.png || { echo "Missing 128x128 icon in app-info" ; exit 1; }
+          test -f app-info/xmls/com.obsproject.Studio.xml.gz || { echo "Missing com.obsproject.Studio.xml.gz in app-info" ; exit 1; }
+
+      - name: Commit screenshots to the OSTree repository
+        run: |
+          ostree commit --repo=repo --canonical-permissions --branch=screenshots/x86_64 flatpak_app/screenshots
```

Comment:
Not really, it must be the toplevel folder. I'm leaving it out for this step.

Reply:
I will note that we need to export an env variable for the `flatpak_app` folder in case we end up renaming it for whatever reasons in flatpak-github-actions